sky/packages/sky/lib/src/widgets/framework.dart

```diff
@@ -626,6 +649,16 @@ abstract class Element<T extends Widget> implements BuildContext {
     _slot = newSlot;
   }
 
+  void _updateDepth() {
+    int expectedDepth = _parent.depth + 1;
```

Comment:
we should just increment the depth to the parent+1 if our depth is <= the parent depth.